openmdao/solvers/solver.py

```diff
@@ -236,6 +236,17 @@ def _convergence_failure(self):
                f"in {self._iter_count} iterations.")
         self.report_failure(msg)
 
+    def can_solve_implicit(self):
+        """
```

Comment:
Nonlinear Block Gauss Seidel (and by extension, probably Linear BGS) can solve cycles, but can't solve implicit components. Might need to split this into 2 checks, can_solve_implcit and can_solve_cycles.

Reply:
I changed can_solve_implicit to can_solve_cycles and use supports['implicit_components'] to check for implicit support since we were already using it in a few places.